server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java

```diff
@@ -317,8 +326,16 @@ public void childEvent(
     catch (Exception ex) {
       throw ex;
     }
-    Assert.assertTrue(assignSegmentLatch.await(4, TimeUnit.SECONDS));
+
+    Assert.assertTrue(assignSegmentLatch.await(2, TimeUnit.SECONDS));
+    Assert.assertEquals(ImmutableMap.of(dataSource, 100.0), coordinator.getLoadStatus());
     curator.delete().guaranteed().forPath(ZKPaths.makePath(LOADPATH, dataSegment.getIdentifier()));
+    // Wait for coordinator thread to run so that replication status is updated
+    // Waiting 50 ms more than the Coordination period should be sufficient
+    Thread.sleep(COORDINATOR_PERIOD + 50);
```

Comment:
If there is no way to do this with latches, then I think it is fine to put the check inside a while loop that sleeps every 50ms and times out at 1s.